pkg/executor/api.go

```diff
@@ -153,9 +154,12 @@ func (executor *Executor) getServiceForFunction(ctx context.Context, fn *fv1.Fun
 		respChan: respChan,
 	}
 	resp := <-respChan
-	if resp.err != nil {
-		return "", resp.err
+	if errors.Is(ctx.Err(), context.Canceled) {
+		et := executor.executorTypes[resp.funcSvc.Executor]
+		et.UnTapService(ctx, crd.CacheKey(resp.funcSvc.Function), resp.funcSvc.Address)
```

Comment:
We should check resp.funcSvc is not nil and then only execute this code as it is a pointer.

Reply:
done